src/test/java/juja/sqlcmd/DatabaseManagerTest.java

```diff
@@ -48,65 +65,109 @@ public void testConnectionIfWrongPassword() {
-            statement.execute(String.format("DROP DATABASE IF EXISTS \"%s\" ", this.dbName));
-            testConnection.close();
-        }
+    @Test
+    public void testGetTableDataWhenNoTable() {
+        databaseManager.connect(dbName, USER, PASSWORD);
+        DataSet[] expected = new DataSet[0];
+        DataSet[] actual = databaseManager.getTableData("tableNotExist");
+        assertArrayEquals(expected, actual);
     }
 
-    private void createTwoTablesInDatabase(String dbName) throws SQLException, ClassNotFoundException {
-        Connection testConnection = getTestConnection(dbName);
-        try (Statement statement = testConnection.createStatement()) {
-            statement.execute("CREATE TABLE first_table (id SERIAL PRIMARY KEY)");
-            statement.execute("CREATE TABLE second_table (id SERIAL PRIMARY KEY)");
-            testConnection.close();
+    @Test
+    public void testGetTableDataWhenNoConnection() {
+        DataSet[] expected = new DataSet[0];
+        DataSet[] actual = databaseManager.getTableData("someTable");
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetTableDataWhenEmptyTable() throws SQLException {
+        String tableName = "some_table";
+        executeQuery(String.format("CREATE TABLE %s (id SERIAL PRIMARY KEY)", tableName));
+        databaseManager.connect(dbName, USER, PASSWORD);
+        DataSet[] expected = new DataSet[0];
+        DataSet[] actual = databaseManager.getTableData(tableName);
+        executeQuery(String.format("DROP TABLE IF EXISTS %s", tableName));
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetTableData() throws SQLException {
+        String tableName = "some_table";
+        createTableWithData(tableName);
+        DataSet firstRow = createDataSet(new String[]{"1", "name1", "25"});
+        DataSet secondRow = createDataSet(new String[]{"2", "name2", "35"});
+        DataSet thirdRow = createDataSet(new String[]{"3", "name3", "45"});
+        databaseManager.connect(dbName, USER, PASSWORD);
+        DataSet[] expected = new DataSet[]{firstRow, secondRow, thirdRow};
+        DataSet[] actual = databaseManager.getTableData(tableName);
+        executeQuery(String.format("DROP TABLE IF EXISTS %s", tableName));
+        assertArrayEquals(expected, actual);
+
+    }
+
+    private DataSet createDataSet(String[] row) {
+        DataSet oneRow = new DataSet(row.length);
+        for (int i = 0; i < row.length; i++) {
+            oneRow.add(i, row[i]);
         }
+        return oneRow;
     }
 
-    private void preparationEnvironment() throws ClassNotFoundException, SQLException {
-        this.dbName = new Date().toString();
-        Connection testConnection = getTestConnection();
-        try (Statement statement = testConnection.createStatement()) {
-            statement.execute(String.format("CREATE DATABASE \"%s\" ", this.dbName));
-            testConnection.close();
+    @After
```

Comment:
I think that @After and @AfterClass blocks should  go after @Before and @BeforeClass. With this blocks you describe a part of test-environment logic and methods annotating with @After, @AfterClass, @Before and @BeforeClass have a strong relation between each other

Reply:
Thanks for your comment, maybe you're right.
I'll think about it